src/hooks/on-create.ts

```diff
@@ -0,0 +1,24 @@
+import { readFileSync, writeFileSync } from 'fs'
+import path from 'path'
+
+type Hook = (args: { projectName: string; directoryPath: string }) => void
```

Comment:
This type definition should not be placed on this file. It's good to create `hooks/types.ts`.